### PR TITLE
mailmap: fixed (all?) author names inconsistencies

### PR DESCRIPTION
**Pull request by kartben**
Ran Levensthein diff on all author names to spot and fix inconsistencies. Used authors' most recently used name and emails as canonical.

This effectively removes 40+ phantom authors, like such as the infamous Marti Bolivar, Martí Bolívar's infamous doppelgänger. @mbolivar

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -35,3 +35,58 @@ Martin Jäger <martin@libre.solar>  <17674105+martinjaeger@users.noreply.github.
 Armand Ciejak <armand@riedonetworks.com>  <armandciejak@users.noreply.github.com>
 Chunlin Han <chunlin.han@linaro.org> <chunlin.han@acer.com>
 chao an <anchao@xiaomi.com>
+Martí Bolívar <marti.bolivar@nordicsemi.no>
+Martí Bolívar <marti.bolivar@nordicsemi.no> <marti.bolivar@linaro.org>
+Martí Bolívar <marti.bolivar@nordicsemi.no> <marti.f.bolivar@gmail.com>
+Martí Bolívar <marti.bolivar@nordicsemi.no> <marti@foundries.io>
+Martí Bolívar <marti.bolivar@nordicsemi.no> <marti@opensourcefoundries.com>
```
Comment (nashif) [inline .mailmap:42]: wow, this guy has lots of identities :-D

Comment (mbolivar-nordic) [reply .mailmap:42]: He can't seem to hold on to a job

**Review by mbolivar-nordic — DISMISSED** [review]

**Comment by kartben** [comment]
@nashif does this have your +1? Thanks!

**Review by nashif — DISMISSED** [review]

**Comment by kartben** [comment]
@nashif sorry, just force-pushed one last omission before seeing you had +1'd :/

**Review by mbolivar-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**